qiling/loader/elf.py

```diff
@@ -414,10 +418,19 @@ def lkm_get_init(self, elffile: ELFFile) -> int:
         raise QlErrorELFFormat('invalid module: symbol init_module not found')
 
     def lkm_dynlinker(self, elffile: ELFFile, mem_start: int) -> Mapping[str, int]:
+        self._symbol_name_map = None
         def __get_symbol(name: str) -> Optional[Symbol]:
             _symtab = elffile.get_section_by_name('.symtab')
-            _sym = _symtab.get_symbol_by_name(name)
 
+            # Cache
+            if self._symbol_name_map == None:
```

Comment:
This could be simplified with an internal function decorated with `@cache` (available from Python 3.9) or `@lru_cache` (available earlier).

```python
from functools import cache

@cache
def __get_cached_symbol(name: str) -> int:
    # access symtab symbol here and return the result
    ...
```

Also, I am not sure that `_symbol_name_map` should be assigned to `self`, but stay local.